# Resolução dos Exercícios Dev Pro

Nesse notebook consta a resolução de varios exercícios da [lista da Python Brasil](https://wiki.python.org.br/ListaDeExercicios).

# [Estrutura Sequencial](https://wiki.python.org.br/EstruturaSequencial)
  ✔️ Concluído.

**Exercício 1.**
* Faça um Programa que mostre a mensagem "Alo mundo" na tela.

In [ ]:
print ('Olá Mundo')

Olá Mundo


**Exercício 2.**
  * Faça um Programa que peça um número e então mostre a mensagem O número informado foi [número].

In [ ]:
numero = input ('Digite um número: ')
print (f'O número informado foi {numero}.')

Digite um número: 10
O número informado foi 10.


*obs: objetos entra abertura de aspas simples significam tipo STR.*

**Exercício 3.**
* Faça um Programa que peça dois números e imprima a soma.

In [ ]:
numero1 = int(input ('Digite um número: '))
numero2 = int(input ('Digite outro número: '))
soma = (numero1 + numero2)
print (f'A soma dos dois números é: {soma}')

Digite um número: 12
Digite outro número: 12
A soma dos dois números é: 24


**Exercício 4.**

* Faça um Programa que peça as 4 notas bimestrais e mostre a média.

In [ ]:
nota1 = eval(input ('Digite a nota 1 do 1° mês: '))
nota2 = eval(input ('Digite a nota 2 do 1° mês: '))
nota3 = eval(input ('Digite a nota 1 do 2° mês: '))
nota4 = eval(input ('Digite a nota 2 do 2° mês: '))
media = int(nota1 + nota2 + nota3 + nota4)
print (f'Sua média bimestral foi: {media/4}')

Digite a nota 1 do 1° mês: 10
Digite a nota 2 do 1° mês: 5.75
Digite a nota 1 do 2° mês: 8
Digite a nota 2 do 2° mês: 6.5
Sua média bimestral foi: 7.5


**Exercício 5.**
* Faça um Programa que converta metros para centímetros.


In [ ]:
metros = eval(input ('Digite o valor em metros para a conversão em cm: '))
print (f'O valor em cm é: {metros * 100}cm')

Digite o valor em metros para a conversão em cm: 2.5
O valor em cm é: 250.0cm


**Exercício 6.**
* Faça um Programa que peça o raio de um círculo, calcule e mostre sua área.

In [ ]:
import math
print ('Calculadora de Círculo')
raio = eval (input ('Digite o raio do circulo para saber sua área: '))
print (f'A área do círculo é: {int(math.pi * raio **2)} (cm2/m2)')

Calculadora de Círculo
Digite o raio do circulo para saber sua área: 50
A área do círculo é: 7853 (cm2/m2)


**Exercício 7.**
 
* Faça um Programa que calcule a área de um quadrado, em seguida mostre o dobro desta área para o usuário.

In [ ]:
print ('Calculadora de Área do Quadrado')
comp = eval (input('Digite o valor do comprimento do quadrado (m/cm): '))
larg = eval (input('Digite o valor da largura do quadrado (m/cm): '))
area = (comp * larg)
print (f'A área do quadrado é {area} (cm2/m2), sendo o dobro dessa área {area*2}(cm2/m2).')

Calculadora de Área do Quadrado
Digite o valor do comprimento do quadrado (m/cm): 10
Digite o valor da largura do quadrado (m/cm): 15
A área do quadrado é 150 (cm2/m2), sendo o dobro dessa área 300(cm2/m2).


**Exercício 8.**

* Faça um Programa que pergunte quanto você ganha por hora e o número de horas trabalhadas no mês. Calcule e mostre o total do seu salário no referido mês.


In [ ]:
print ('Descubra o seu salario em horas e dias trabalhados.')
shora = eval( input ('Digite quanto você ganha em 01 hora trabalhada: '))
dhora = eval( input ('Digite quantas horas você trabalha em 01 dia: '))
dias_semana = int( input ('Digite quantos dias você trabalhou em 01 semana (ex.: 5): '))
semana_mes = int( input ('Quantas semanas você trabalhou em 01 mês (ex.: 4): '))
folgas_mes = int( input ('Digite quantas faltas (descontadas) você teve em 01 mês (ex.: 2): '))
diaria = int( shora * dhora)
salario = (diaria * dias_semana * semana_mes)
print (f'Seu salário foi de {salario - (shora * folgas_mes)},00 reais')

Descubra o seu salario em horas e dias trabalhados.
Digite quanto você ganha em 01 hora trabalhada: 25
Digite quantas horas você trabalha em 01 dia: 8
Digite quantos dias você trabalhou em 01 semana (ex.: 5): 5
Quantas semanas você trabalhou em 01 mês (ex.: 4): 4
Digite quantas faltas (descontadas) você teve em 01 mês (ex.: 2): 2
Seu salário foi de 3950,00 reais


**Exercício 9.**
* Faça um Programa que peça a temperatura em graus Fahrenheit, transforme e mostre a temperatura em graus Celsius.
C = 5 * ((F-32) / 9).

In [ ]:
print ('Conversor de Fahrenheit em Celsius')
fah = float(input ("Digite o valor em Fahrenheit para a conversão em Celsius: "))
celsius = (5 * ((fah-32) / 9))
print (f"O valor em Celsius é {celsius:.2f}°C.")

Conversor de Fahrenheit em Celsius
Digite o valor em Fahrenheit para a conversão em Celsius: 79
O valor em Celsius é 26.11°C.


**Exercício 10.**
* Faça um Programa que peça a temperatura em graus Celsius, transforme e mostre em graus Fahrenheit.

In [ ]:
print ('Conversor de Celsius em Fahrenheit')
celsius = float(input ("Digite o valor em Celsius para a conversão em Fahrenheit: "))
fah = (celsius * 1.8) + 32
print (f"O valor em Fahrenheit é {fah}°F.")

Conversor de Celsius em Fahrenheit
Digite o valor em Celsius para a conversão em Fahrenheit: 25
O valor em Fahrenheit é 77.0°F.


**Exercício 11.**
* Faça um Programa que peça 2 números inteiros e um número real. 
Calcule e mostre:

* o produto do dobro do primeiro com metade do segundo.
* a soma do triplo do primeiro com o terceiro.
* o terceiro elevado ao cubo.

In [ ]:
print ('Calculando no Python. Para isso, precisaremos de 03 números')
numero1 = int(input ('Digite o 1° número inteiro: '))
numero2 = int(input ('Digite o 2° número inteiro: '))
numero3 = float(input ('Agora digite um número real: '))
calculo1 = (numero1*2) * (numero2/2)
calculo2 = (numero1*3) + numero3
calculo3 = numero3**3
print (f'O produto do dobro do primeiro com metado do segundo é {calculo1}')
print (f'A soma do triplo do primeiro com o terceiro é {calculo2}')
print (f'O número real elevado ao cubo é {calculo3}')

Calculando no Python. Para isso, precisaremos de 03 números
Digite o 1° número inteiro: 50
Digite o 2° número inteiro: 32
Agora digite um número real: 75.80
O produto do dobro do primeiro com metado do segundo é 1600.0
A soma do triplo do primeiro com o terceiro é 225.8
O número real elevado ao cubo é 435519.51199999993


**Exercício 12.**
* Tendo como dados de entrada a altura de uma pessoa, construa um algoritmo que calcule seu peso ideal, usando a seguinte fórmula: (72.7*altura) - 58

In [ ]:
print ('Saiba seu peso ideal com base na sua altura:')
altura1 = float( input('Informe-nos sua altura: '))
peso = float( (72.7 * altura1) - 58)
print (f'Seu peso ideal seria: {peso:.2f}')


Saiba seu peso ideal com base na sua altura:
Informe-nos sua altura: 1.69
Seu peso ideal seria: 64.86


**Exercício 13.**
* Tendo como dado de entrada a altura (h) de uma pessoa, construa um algoritmo que calcule seu peso ideal, utilizando as seguintes fórmulas:
  * Para homens: (72.7*h) - 58
  * Para mulheres: (62.1*h) - 44.7

In [ ]:
print ('Saiba seu peso ideal com base na sua altura!')
answer = (input ('Para começarmos informe-nos seu sexo (homem/mulher): '))
answer = answer.lower()
if answer == "homem":
  altura = float( input('Informe-nos sua altura: '))
  peso = float( (72.7 * altura) - 58)
  print (f'Seu peso ideal seria: {peso:.2f}kgs')

if answer == "mulher":
  altura = float( input('Informe-nos sua altura: '))
  peso = float( (62.1 * altura) - 44.7)
  print (f'Seu peso ideal seria: {peso:.2f}kgs')

Saiba seu peso ideal com base na sua altura!
Para começarmos informe-nos seu sexo (homem/mulher): HOmEm
Informe-nos sua altura: 1.87
Seu peso ideal seria: 77.95kgs


**Exercício 14.**
* João Papo-de-Pescador, homem de bem, comprou um microcomputador para controlar o rendimento diário de seu trabalho. Toda vez que ele traz um peso de peixes maior que o estabelecido pelo regulamento de pesca do estado de São Paulo (50 quilos) deve pagar uma multa de R$ 4,00 por quilo excedente. **João precisa que você faça um programa que leia a variável peso (peso de peixes) e calcule o excesso.** Gravar na variável excesso a quantidade de quilos além do limite e na variável multa o valor da multa que João deverá pagar. Imprima os dados do programa com as mensagens adequadas.

In [ ]:
print ('Vamos saber o tamanho do prejuízo do João Papo-de-Pescador!')
peixes = float( input('João, quantos kilos de peixe você pescou hoje? '))
excedente = float(peixes - 50)
print (f'Tendo em vista que você excedeu {excedente}kgs, sua multa será no valor de R${excedente * 4:.2f} .')

Vamos saber o tamanho do prejuízo do João Papo-de-Pescador!
João, quantos kilos de peixe você pescou hoje? 217.82
Tendo em vista que você excedeu 167.82kgs, sua multa será no valor de R$671.28 .


**Exercício 15.**
* Faça um Programa que pergunte quanto você ganha por hora e o número de horas trabalhadas no mês. Calcule e mostre o total do seu salário no referido mês, sabendo-se que são descontados 11% para o Imposto de Renda, 8% para o INSS e 5% para o sindicato, faça um programa que nos dê:

  a) salário bruto.
  
  b) quanto pagou ao INSS.
  
  c) quanto pagou ao sindicato.
  
  d) o salário líquido.

  * Calcule os descontos e o salário líquido, conforme a tabela abaixo:
    * Salário Bruto : R$
    * IR (11%) : R$
    * INSS (8%) : R$
    * Sindicato ( 5%) : R$
    * Salário Liquido : R$
    
Obs.: Salário Bruto - Descontos = Salário Líquido

In [ ]:
print ('Holerite Eletrônico Mensal')
hora_t = float( input('Informe quanto você ganha por hora trabalhada: '))
hora_mes = float( input('Informe quantas horas você trabalha no mês: '))
print ('De acordo com as informações fornecidas:')
salario = (hora_t * hora_mes)
ir = (salario * 11/100)
inss = (salario * 8/100)
sindicato = (salario * 5/100)
descontos = (ir + inss + sindicato)
print (f'Seu salário bruto foi de: R$ {salario:.2f}')
print (f'Contribuição de IR(11%): R$ {ir:.2f}')
print (f'Contribuição de INSS(8%): R$ {inss:.2f}')
print (f'Contribuição ao Sindicato(5%): R$ {sindicato:.2f}')
print (f'Valor do Salário Liquido: R$ {salario - descontos:.2f}')

Holerite Eletrônico Mensal
Informe quanto você ganha por hora trabalhada: 25
Informe quantas horas você trabalha no mês: 160
De acordo com as informações fornecidas:
Seu salário bruto foi de: R$ 4000.00
Contribuição de IR(11%): R$ 440.00
Contribuição de INSS(8%): R$ 320.00
Contribuição ao Sindicato(5%): R$ 200.00
Valor do Salário Liquido: R$ 3040.00


**Exercício 16.**
* Faça um programa para uma loja de tintas. 

O programa deverá pedir o tamanho em metros quadrados da área a ser pintada.

Considere que a cobertura da tinta é de 1 litro para cada 3 metros quadrados e que a tinta é vendida em latas de 18 litros, que custam R$ 80,00. 

* Informe ao usuário a quantidades de latas de tinta a serem compradas e o preço total.


In [ ]:
import math

print ('Olá, vamos descobrir quantas latas de tinta você precisa!')
area_a_ser_p = int( input ('Digite o tamanho da área a ser pintada em m²: '))

litros_p_metro = area_a_ser_p / 3
litros_por_lata = 18
numero_de_latas = litros_p_metro / litros_por_lata
# print (litros_p_metro , numero_de_latas)
latas_a_serem_compradas = math.ceil (numero_de_latas)
custo_latas = latas_a_serem_compradas * 80

print (f'Você precisará de {latas_a_serem_compradas} latas. Total da compra: R$ {custo_latas:.2f}.')

Olá, vamos descobrir quantas latas de tinta você precisa!
Digite o tamanho da área a ser pintada em m²: 24
Você precisará de 1 latas. Total da compra: R$ 80.00.


**Exercício 17.**

Faça um Programa para uma loja de tintas. O programa deverá pedir o tamanho em metros quadrados da área a ser pintada. Considere que a cobertura da tinta é de 1 litro para cada 6 metros quadrados e que a tinta é vendida em latas de 18 litros, que custam 80 reais ou em galões de 3,6 litros, que custam 25 reais.

Informe ao usuário as quantidades de tinta a serem compradas e os respectivos preços em 3 situações:

* comprar apenas latas de 18 litros;
* comprar apenas galões de 3,6 litros;
* misturar latas e galões, de forma que o desperdício de tinta seja menor.
* Acrescente 10% de folga e sempre arredonde os valores para cima, isto é, considere latas cheias.

In [ ]:
import math

print ('Olá, vamos descobrir quantas latas de tinta você precisa!')
area_a_ser_p = float( input ('Digite o tamanho da área a ser pintada em m²: '))
area_com_folga = area_a_ser_p * 1.1

# Para latas de 18L por R$80
litros_por_metro = 6
litros_a_serem_usados = area_com_folga / litros_por_metro
litros_por_lata = 18
numero_de_latas = math.ceil (litros_a_serem_usados / litros_por_lata)
custo_com_apenas_latas = numero_de_latas * 80
print (f'Você precisará de {numero_de_latas} latas de 18 litros. Total da compra: R$ {custo_com_apenas_latas:.2f}.')

# Para galões de 3,6L por R$25
litros_por_galao = 3.6
numero_de_galoes = math.ceil (litros_a_serem_usados / litros_por_galao)
custo_com_apenas_galoes = numero_de_galoes * 25
print (f'Você precisará de {numero_de_galoes} galões de 3.6 litros. Total da compra: R$ {custo_com_apenas_galoes:.2f}.')

# Compra de tinta otimizada por valor
numero_de_latas = math.floor (litros_a_serem_usados / litros_por_lata)
valor_de_latas = numero_de_latas * 80
# resto = latas_a_serem_compradas % litros_por_lata
litros_faltantes = litros_a_serem_usados % litros_por_lata

numero_de_galoes = math.ceil (litros_faltantes / litros_por_galao)
valor_de_galoes = numero_de_galoes * 25

valor_total = valor_de_latas + valor_de_galoes
print (f'Você precisará de {numero_de_latas} latas de 18 litros mais {numero_de_galoes} de 3.6 litros. Total da compra: R$ {valor_total:.2f}.')

Olá, vamos descobrir quantas latas de tinta você precisa!
Digite o tamanho da área a ser pintada em m²: 200
Você precisará de 3 latas de 18 litros. Total da compra: R$ 240.00.
Você precisará de 11 galões de 3.6 litros. Total da compra: R$ 275.00.
Você precisará de 2 latas de 18 litros mais 1 de 3.6 litros. Total da compra: R$ 185.00.


**Exercício 18.**
* Faça um programa que peça o tamanho de um arquivo para download (em MB) e a velocidade de um link de Internet (em Mbps), calcule e informe o tempo aproximado de download do arquivo usando este link (em minutos).


In [ ]:
arquivo = float(input("Informe do tamanho do arquivo em MegaByte: "))
link = float(input("Informe a velocidade do link em Mbps: "))
tempo = ((arquivo * 8) / link) / 60
print ("O tempo aproximado de download é de %.2f minutos" %tempo)

Informe do tamanho do arquivo em MegaByte: 100
Informe a velocidade do link em Mbps: 250
O tempo aproximado de download é de 0.05 minutos


# [Estrutura de Decisão](https://wiki.python.org.br/EstruturaDeDecisao)
✔️ Concluído.

**Exercício 1.**
* Faça um Programa que peça dois números e imprima o maior deles.

In [ ]:
a = float( input('Insira um número: '))
b = float( input('Insira outro número: '))
if a > b:
  print (f'O maior número é o número {a}')
if a < b:
  print (f'O maior número é o número {b}')
if a==b:
  print ('Os números são iguais.')


Insira um número: 45
Insira outro número: 45.57
O maior número é o número 45.57


**Exercício 2.**
* Faça um Programa que peça um valor e mostre na tela se o valor é positivo ou negativo.


In [ ]:
a = float( input('Insira um número: '))
if a > 0:
  print ('O numero informado é positivo.')
elif a < 0:
  print ('O numero informado é negativo.')
else:
  print ('O numero informado é igual a 0.')

Insira um número: 0
O numero informado é igual a 0.


**Exercício 3.**
* Faça um Programa que verifique se uma letra digitada é "F" ou "M". Conforme a letra escrever: F - Feminino, M - Masculino, Sexo Inválido.

In [ ]:
sexo = input('Informe o seu sexo. ("M" para Masculino / "F" para Feminino): ')
sexo = sexo.upper()
if sexo == 'M':
  print ('Você é do sexo masculino.')
elif sexo == 'F':
  print ('Você é do sexo feminino.')
else:
  print ('Sexo Inválido.')

Informe o seu sexo. ("M" para Masculino / "F" para Feminino): m
Você é do sexo masculino.


**Exercício 4.**
* Faça um Programa que verifique se uma letra digitada é vogal ou consoante.

In [ ]:
vogais = ['a', 'e', 'i', 'o', 'u']

letra = input("Digite uma letra: ")
letra = letra.lower()
if letra in vogais:
  print ("Você digitou uma vogal.")
elif letra.isalpha():
  print ("Você digitou uma consoante.")
else:
  print ("O que você digitou não é uma letra.")

Digite uma letra: 10
O que você digitou não é uma letra.


**Exercício 5.**
* Faça um programa para a leitura de duas notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e apresentar:
    * A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;
    * A mensagem "Reprovado", se a média for menor do que sete;
    * A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [ ]:
print ('Checagem de aprovação para o próximo grau de ensino.')
aluno = input('Por favor, insira o nome do aluno: ')
nota1 = float( input('Informe a 1ª nota: '))
nota2= float( input('Informe a 2ª nota: '))

if nota1 and nota2 > 10: 
  quit()
  print ('Valor de nota invalido.')
  
media = (nota1+nota2)/2
if media >= 7 and media < 9.75:
  print (f'O aluno {aluno} foi Aprovado. Sua média bimestral foi {media:.2f}.')
if media == 10:
  print (f'O aluno {aluno} foi Aprovado com Distinção. Sua média bimestral foi {media:.2f}.')
if media < 7:
  print (f'O aluno {aluno} infelizmente foi Reprovado. Sua média bimestral foi {media:.2f}.')

Checagem de aprovação para o próximo grau de ensino.
Por favor, insira o nome do aluno: Luiz Carlos
Informe a 1ª nota: 4.5
Informe a 2ª nota: 10
O aluno Luiz Carlos foi Aprovado. Sua média bimestral foi 7.25.


**Exercício 6.**
* Faça um Programa que leia três números e mostre o maior deles.


In [ ]:
print ('Deixe-me descobrir qual o maior número. Insira 03 números distintos nos campos abaixo.')
n_1 = input('Digite o 1º numero: ')
n_2 = input('Digite o 2º numero: ')
n_3 = input('Digite o 3º numero: ')

# Maior numero.
if n_1 > n_2 and n_1 > n_3:
  quit()
  print (f'O número {n_1} é o maior.') 

if n_2 > n_1 and n_2 > n_3:
  quit()
  print (f'O número {n_2} é o maior.') 

if n_3 > n_2 and n_3 > n_1:
  quit()
  print (f'O número {n_3} é o maior.')

# Caso dois numeros forem iguais.
elif n_1 == n_2 or n_1 == n_3:
  quit()
  print (f'Obs: O número {n_1} foi informado mais de 01 vez.')

elif n_2 == n_1 or n_2 == n_3:
  print (f'Obs: O número {n_2} foi informado mais de 01 vez.') 

elif n_3 == n_2 or n_3 == n_1:
  print (f'Obs: O número {n_3} foi informado mais de 01 vez.') 


Deixe-me descobrir qual o maior número.
Digite o 1º numero: 145.45
Digite o 2º numero: 145.75
Digite o 3º numero: 145.43
O número 145.75 é o maior.


**Excercício 7.**
* Faça um Programa que leia três números e mostre o maior e o menor deles.


In [ ]:
print ('Deixe-me descobrir qual o maior e o menor número. Insira 03 números distintos nos campos abaixo.')
n_1 = input('Digite o 1º numero: ')
n_2 = input('Digite o 2º numero: ')
n_3 = input('Digite o 3º numero: ')

# Maior numero.
if n_1 > n_2 and n_1 > n_3:
  quit()
  print (f'O número {n_1} é o maior.') 

if n_2 > n_1 and n_2 > n_3:
  quit()
  print (f'O número {n_2} é o maior.') 

if n_3 > n_2 and n_3 > n_1:
  quit()
  print (f'O número {n_3} é o maior.')

# Menor numero.
if n_1 < n_2 and n_1 < n_3:
  quit()
  print (f'O número {n_1} é o menor.') 

if n_2 < n_1 and n_2 < n_3:
  quit()
  print (f'O número {n_2} é o menor.') 

if n_3 < n_2 and n_3 < n_1:
  quit()
  print (f'O número {n_3} é o menor.')


# Caso dois numeros forem iguais.
elif n_1 == n_2 or n_1 == n_3:
  quit()
  print (f'Obs: O número {n_1} foi informado mais de 01 vez.')

elif n_2 == n_1 or n_2 == n_3:
  print (f'Obs: O número {n_2} foi informado mais de 01 vez.') 

elif n_3 == n_2 or n_3 == n_1:
  print (f'Obs: O número {n_3} foi informado mais de 01 vez.') 

Deixe-me descobrir qual o maior e o menor número. Insira 03 números distintos nos campos abaixo.
Digite o 1º numero: 57.78
Digite o 2º numero: 57.77
Digite o 3º numero: 57
O número 57.78 é o maior.
O número 57 é o menor.


**Exercício 8.**
* Faça um programa que pergunte o preço de três produtos e informe qual produto você deve comprar, sabendo que a decisão é sempre pelo mais barato.


In [ ]:
print ('Seletor de produto mais econômico. Qual o valor dos 03 produtos abaixo?')
prod1 = float( input ('LG Smart TV LED 4k 65 polegadas: R$ '))
prod2 = float( input ('LG Smart TV UltraHD 4k 65 polegadas (Americanas): R$ '))
prod3 = float( input ('LG Smart TV UltraHD 4k 65 polegadas (Magazine Luiza): R$ '))

# TV mais barata.
if prod1 < prod2 and prod1 < prod3:
  quit()
  print (f'O produto LG Smart TV LED 4k 65 polegadas, no valor de R$ {prod1:.2f} é o mais em conta entre as 03 opções.') 

if prod2 < prod1 and prod2 < prod3:
  quit()
  print (f'O produto LG Smart TV UltraHD 4k 65 polegadas (Americanas), no valor de R$ {prod2:.2f} é o mais em conta entre as 03 opções.') 

if prod3 < prod2 and prod3 < prod1:
  quit()
  print (f'O produto LG Smart TV UltraHD 4k 65 polegadas (Magazine Luiza), no valor de  R$ {prod3:.2f} é o mais em conta entre as 03 opções.')

Seletor de produto mais econômico. Qual o valor dos 03 produtos abaixo?
LG Smart TV LED 4k 65 polegadas: R$ 4523
LG Smart TV UltraHD 4k 65 polegadas (Americanas): R$ 20000
LG Smart TV UltraHD 4k 65 polegadas (Magazine Luiza): R$ 3999
O produto LG Smart TV UltraHD 4k 65 polegadas (Magazine Luiza), no valor de  R$ 3999.00 é o mais em conta entre as 03 opções.


**Exercício 9.**
* Faça um Programa que leia três números e mostre-os em ordem decrescente.


In [ ]:
lista = []
numeros = 3
for i in range(numeros):
    elemento = eval(input('Insira um numero: '))
    lista.append(elemento)

lista.sort(reverse = True) # ativa ordem decrescente dos elementos
print(lista)

Insira um numero: 587
Insira um numero: 888.1
Insira um numero: 888.0
[888.1, 888.0, 587]


**Exercício 10.**
* Faça um Programa que pergunte em que turno você estuda. Peça para digitar M-matutino ou V-Vespertino ou N- Noturno. Imprima a mensagem "Bom Dia!", "Boa Tarde!" ou "Boa Noite!" ou "Valor Inválido!", conforme o caso.


In [ ]:
print ('Olá! Qual turno você estuda?')
turno = input('Digite "M" para Manhã / "V" para Vespertino / "N" para Noturno: ')
turno = turno.upper()
if turno == 'M':
  print ('Bom dia e bons estudos!!')
elif turno == 'V':
  print ('Boa tarde e bons estudos!!')
elif turno == 'N':
  print ('Boa noite e bons estudos!!')
else:
  print ('Valor Inválido.')

Olá! Qual turno você estuda?
Digite "M" para Manhã / "V" para Vespertino / "N" para Noturno: M
Bom dia e bons estudos!!


**Exercício 11.**
* As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contraram para desenvolver o programa que calculará os reajustes.

* Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:
    * salários até R$ 280,00 (incluindo) : aumento de 20%
    * salários entre R$ 280,00 e R$ 700,00 : aumento de 15%
    * salários entre R$ 700,00 e R$ 1500,00 : aumento de 10%
    * salários de R$ 1500,00 em diante : aumento de 5%.

Após o aumento ser realizado, informe na tela:
* o salário antes do reajuste;
* o percentual de aumento aplicado;
* o valor do aumento;
* o novo salário, após o aumento.

In [ ]:
print ('Calculadora de Aumento de Salário das Organizações Tabajara.')
salario = float(input('Por favor, digite o valor do seu salário mensal: '))

vinte_porcento = salario * 20/100
quinze_porcento = salario * 15/100
dez_porcento = salario * 10/100
cinco_porcento = salario * 5/100

if salario <= 280:
  print (f'Conforme informado, seu salario era de R$ {salario:.2f}.')
  print (f'Logo, o percentual de reajuste foi de 20%.')
  print (f'Sendo assim, o valor do aumento salarial corresponde a R$ {vinte_porcento:.2f}.')
  print (f'O valor do seu novo salário é de R$ {salario + vinte_porcento:.2f}.')

elif salario > 280 and salario <= 700:
  print (f'Conforme informado, seu salario era de R$ {salario:.2f}.')
  print (f'Logo, o percentual de reajuste foi de 15%.')
  print (f'Sendo assim, o valor do aumento salarial corresponde a R$ {quinze_porcento:.2f}.')
  print (f'O valor do seu novo salário é de R$ {salario + quinze_porcento:.2f}.')

elif salario > 700 and salario <= 1500:
  print (f'Conforme informado, seu salario era de R$ {salario:.2f}.')
  print (f'Logo, o percentual de reajuste foi de 10%.')
  print (f'Sendo assim, o valor do aumento salarial corresponde a R$ {dez_porcento:.2f}.')
  print (f'O valor do seu novo salário é de R$ {salario + dez_porcento:.2f}.')

if salario > 1500:
  print (f'Conforme informado, seu salario era de R$ {salario:.2f}.')
  print (f'Logo, o percentual de reajuste foi de 05%.')
  print (f'Sendo assim, o valor do aumento salarial corresponde a R$ {cinco_porcento:.2f}.')
  print (f'O valor do seu novo salário é de R$ {salario + cinco_porcento:.2f}.')


Calculadora de Aumento de Salário das Organizações Tabajara.
Por favor, digite o valor do seu salário mensal: 584
Conforme informado, seu salario era de R$ 584.00.
Logo, o percentual de reajuste foi de 15%.
Sendo assim, o valor do aumento salarial corresponde a R$ 87.60.
O valor do seu novo salário é de R$ 671.60.


**Exercício 12.**
* Faça um programa para o cálculo de uma folha de pagamento, sabendo que os descontos são do Imposto de Renda, que depende do salário bruto (conforme tabela abaixo) que o INSS corresponde a 10% do Salário Bruto, tal como o FGTS a 11% (obs: o FGTS não é descontado, é a empresa que deposita). 

O Salário Líquido corresponde ao Salário Bruto menos os descontos. O programa deverá pedir ao usuário o valor da sua hora e a quantidade de horas trabalhadas no mês.

Desconto do IR:
* Salário Bruto até 900 (inclusive) - isento
* Salário Bruto até 1500 (inclusive) - desconto de 5%
* Salário Bruto até 2500 (inclusive) - desconto de 10%
* Salário Bruto acima de 2500 - desconto de 20% 

Imprima na tela as informações, dispostas conforme o exemplo abaixo. 

No exemplo o valor da hora é 5 e a quantidade de hora é 220.


        Salário Bruto: (5 * 220)        : R$ 1100,00
        (-) IR (5%)                     : R$   55,00  
        (-) INSS ( 10%)                 : R$  110,00
        FGTS (11%)                      : R$  121,00
        Total de descontos              : R$  165,00
        Salário Liquido                 : R$  935,00


In [ ]:
print ('Sistema de Folha de Pagamento - Phoenix Mercantil Ltda.')
vhora = float(input('Por favor, informe o valor da sua hora trabalhada: '))
htrab = int(input('Quantas hora você trabalhou neste mês? '))
salario_bruto = (vhora * htrab)
inss = (salario_bruto * 10/100)
fgts = (salario_bruto * 11/100)
ir = 0
descontos = (inss + ir)
salario_liquido = (salario_bruto - descontos)

# Isenção até R$900
if salario_bruto <= 900:
  print (f'Salário Bruto: ( {vhora:.2f} * {htrab})           : R$ {salario_bruto:.2f}')
  print (f'(-) IR   (0%)                          : Isento ')
  print (f'(-) INSS (10%)                         : R$ {inss:.2f}')
  print (f'FGTS     (11%)                         : R$ {fgts:.2f}')
  print (f'Total de descontos                     : R$ {descontos:.2f}')
  print (f'Salário Líquido                        : R$ {salario_liquido:.2f}')

# IR: 5% , salário de R$901 até R$1500
elif salario_bruto > 900 and salario_bruto <= 1500:
  ir = (salario_bruto * 5/100)
  descontos = (inss + ir)
  salario_liquido = (salario_bruto - descontos)
  print (f'Salário Bruto: ({vhora:.2f} * {htrab}) c           : R$ {salario_bruto:.2f}')
  print (f'(-) IR   (5%)                           : R$ {ir:.2f}')
  print (f'(-) INSS (10%)                          : R$ {inss:.2f}')
  print (f'FGTS     (11%)                          : R$ {fgts:.2f}')
  print (f'Total de descontos                      : R$ {descontos:.2f}')
  print (f'Salário Líquido                         : R$ {salario_liquido:.2f}')

# IR: 10% salário de R$1501 até 2500
elif salario_bruto > 1500 and salario_bruto <= 2500:
  ir = (salario_bruto * 10/100)
  descontos = (inss + ir)
  salario_liquido = (salario_bruto - descontos)
  print (f'Salário Bruto: ({vhora:.2f} * {htrab})             : R$ {salario_bruto:.2f}')
  print (f'(-) IR   (10%)                          : R$ {ir:.2f}')
  print (f'(-) INSS (10%)                          : R$ {inss:.2f}')
  print (f'FGTS     (11%)                          : R$ {fgts:.2f}')
  print (f'Total de descontos                      : R$ {descontos:.2f}')
  print (f'Salário Líquido                         : R$ {salario_liquido:.2f}')

# IR: 20% salário acima de R$2501
if salario_bruto > 2500:
  ir = (salario_bruto * 20/100)
  descontos = (inss + ir)
  salario_liquido = (salario_bruto - descontos)
  print (f'Salário Bruto: ({vhora:.2f} * {htrab})            : R$ {salario_bruto:.2f}')
  print (f'(-) IR   (20%)                          : R$ {ir:.2f}')
  print (f'(-) INSS (10%)                          : R$ {inss:.2f}')
  print (f'FGTS     (11%)                          : R$ {fgts:.2f}')
  print (f'Total de descontos                      : R$ {descontos:.2f}')
  print (f'Salário Líquido                         : R$ {salario_liquido:.2f}')

Sistema de Folha de Pagamento - Phoenix Mercantil Ltda.
Por favor, informe o valor da sua hora trabalhada: 20
Quantas hora você trabalhou neste mês? 160
Salário Bruto: (20.00 * 160)            : R$ 3200.00
(-) IR   (20%)                          : R$ 640.00
(-) INSS (10%)                          : R$ 320.00
FGTS     (11%)                          : R$ 352.00
Total de descontos                      : R$ 960.00
Salário Líquido                         : R$ 2240.00


** Exercício 13.**
* Faça um Programa que leia um número e exiba o dia correspondente da semana. (1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [ ]:
print ('Numeros de referência quanto aos dias da semana')
numero = int( input('Digite um número de 1 a 7: '))
if numero > 7 or numero < 0:
  print ('Valor Inválido')

if numero == 1:
  print('O número 1 corresponde ao Domingo')

elif numero == 2:
  print('O número 2 corresponde a Segunda-Feira')

elif numero == 3:
  print('O número 3 corresponde a Terça-Feira')

elif numero == 4:
  print('O número 4 corresponde a Quarta-Feira')

elif numero == 5:
  print('O número 5 corresponde a Quinta-Feira')

elif numero == 6:
  print('O número 6 corresponde a Sexta-Feira')

if numero == 7:
  print('O número 1 corresponde ao Sabado')

Numeros de referência quanto aos dias da semana
Digite um número de 1 a 7: 3
O número 3 corresponde a Terça-Feira


**Exercício 14.**
* Faça um programa que lê as duas notas parciais obtidas por um aluno numa disciplina ao longo de um semestre, e calcule a sua média. 

* A atribuição de conceitos obedece à tabela abaixo:

Média de Aproveitamento  Conceito

Entre 9.0 e 10.0        A

Entre 7.5 e 9.0         B

Entre 6.0 e 7.5         C

Entre 4.0 e 6.0         D

Entre 4.0 e zero        E

O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito for D ou E.


In [ ]:
print ('Verificador de Aprovação Semestral')
aluno = input('Digite o nome do aluno: ')
prova1 = float( input('Favor informar a nota da P1: '))
prova2 = float( input('Favor informar a nota da P2: '))
media = (prova1+prova2)/2

if media > 6 and media < 10:
  print (f'O aluno {aluno} obteve a média semestral de {media:.2f}.')
  print ('APROVADO.')

elif media < 6 and media >= 0:
  print (f'O aluno {aluno} obteve a média semestral de {media:.2f}.')
  print ('REPROVADO.')

else:
  print ('Valor Inválido. Tente novamente.')

Verificador de Aprovação Semestral
Digite o nome do aluno: Luiz
Favor informar a nota da P1: 10
Favor informar a nota da P2: 7.5
O aluno Luiz obteve a média semestral de 8.75.
APROVADO.


* Faça um Programa que peça os 3 lados de um triângulo. O programa deverá informar se os valores podem ser um triângulo. Indique, caso os lados formem um triângulo, se o mesmo é: equilátero, isósceles ou escaleno.
Dicas:

Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;

  * Triângulo Equilátero: três lados iguais;
  * Triângulo Isósceles: quaisquer dois lados iguais;
  * Triângulo Escaleno: três lados diferentes;

In [ ]:
lado_a = float( input('Insira o valor do lado A do triangulo: '))
lado_b = float( input('Insira o valor do lado B do triangulo: '))
lado_c = float( input('Insira o valor do lado C do triangulo: ')) 

# Triangulo Check Up
if lado_a < lado_b + lado_c and lado_b < lado_a + lado_c and lado_c < lado_a + lado_b:
  print ('Os valores informados podem formar um triangulo.')
  if lado_a == lado_b == lado_c:
    print ('Triangulo Equilátero.')
  elif lado_a != lado_b != lado_c != lado_a:
    print ('Triangulo Escaleno.')  
  else:
    print ('Triangulo Isóceles.')
else:
  print ('Os valores informados não podem formar um triangulo.')

Insira o valor do lado A do triangulo: 5
Insira o valor do lado B do triangulo: 4
Insira o valor do lado C do triangulo: 5
Os valores informados podem formar um triangulo.
Triangulo Isóceles.


**Exercício 16.**

* Faça um programa que calcule as raízes de uma equação do segundo grau, na forma ax2 + bx + c. O programa deverá pedir os valores de a, b e c e fazer as consistências, informando ao usuário nas seguintes situações:

    * Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e o programa não deve fazer pedir os demais valores, sendo encerrado;

    * Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao usuário e encerre o programa;

    * Se o delta calculado for igual a zero a equação possui apenas uma raiz real; informe-a ao usuário;

    * Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário;

In [ ]:
import math

a = int( input('Digite o valor de A: '))

if a == 0:
  quit()
  print ('Programa Encerrado - A equação não é do segundo grau.')

else:
  b = int (input('Digite o valor de B: '))
  c = int (input('Digite o valor de C: '))
  delta = b*b - (4*a*c)
  
  if delta < 0:
    print ('Programa Encerrado - As equação não possui raizes reais.')

  elif delta == 0:
    raiz = -b / (2*a)
    print ('Delta=0 , raiz = ', raiz)

  else:
    raiz1 = (-b + math.sqrt(delta) ) / (2*a)
    raiz2 = (-b  - math.sqrt(delta) ) / (2*a)
    print (f'Raizes: {raiz1:.2f} e {raiz2:.2f} .')

Digite o valor de A: 5
Digite o valor de B: 10
Digite o valor de C: 4
Raizes: -0.55 e -1.45 .


**Exercício 17.** 
* Faça um Programa que peça um número correspondente a um determinado ano e em seguida informe se este ano é ou não bissexto.


In [ ]:
from calendar import isleap

print ('Verificador de Ano Bissexto.')

ano = int( input('Digite o ano: '))

if isleap(ano):
    print(f'O ano {ano} é bissexto.')
else:
    print(f'O ano {ano} não é bissexto.')

Verificador de Ano Bissexto.
Digite o ano: 2000
O ano 2000 é bissexto.


**Exercício 18.**
* Faça um Programa que peça uma data no formato dd/mm/aaaa e determine se a mesma é uma data válida.


In [ ]:
print ('Olá. Digite uma data para verificar se esta é válida.')

dia = int( input('Digite o dia: '))
mes = int( input('Digite o mês: '))
ano = int( input('Digite o ano: '))

valida = False

# Meses com 31 dias
if mes==1 or mes==3 or mes==5 or mes==7 or mes==8 or mes==10 or mes==12:
  if dia <=31:
    valida = True

# Meses com 30 dias
elif mes==4 or mes==6 or mes==9 or mes==11:
  if dia <= 30:
    valida = True

elif mes==2:
# Teste se é bissexto
  if ano%4==0 and ano%100!=0 or ano%400==0:
    if dia <= 29:
      valida = True
    elif dia <= 28:
      valida = True

if(valida):
  print('Data Válida.')
else:
  print('Data Inválida.')  

Olá. Digite uma data para verificar se esta é válida.
Digite o dia: 31
Digite o mês: 2
Digite o ano: 2020
Data Inválida.


**Exercicio 19.**
* Faça um Programa que leia um número inteiro menor que 1000 e imprima a quantidade de centenas, dezenas e unidades do mesmo.


Observando os termos no plural a colocação do "e", da vírgula entre outros. Exemplo:

* 326 = 3 centenas, 2 dezenas e 6 unidades
* 12 = 1 dezena e 2 unidades Testar com: 326, 300, 100, 320, 310,305, 301, 101, 311, 111, 25, 20, 10, 21, 11, 1, 7 e 16

In [ ]:
import math

print ('Identificador de Centena, Dezena e Unidade')
print (' '*43)

numero = int( input('Digite um número inteiro menor que 1000: '))
if numero < 1000:
# Identificando a Unidade
  unidade = numero % 10
# Eliminando a Unidade 
  numero = (numero - unidade)/10

# Identificando a Dezena
  dezena = numero % 10

# Eliminando a Dezena e Identificando a Centena
  numero = (numero - dezena)/10
  centena = numero

# Transformando os números em Inteiros
  dezena = int(dezena)
  centena = int(centena)
  print(centena,'centena(s),',dezena,'dezena(s) e',unidade,'unidade(s).')

else:
  print ('Número Inválido.')

Identificador de Centena, Dezena e Unidade
                                           
Digite um número inteiro menor que 1000: 324
3 centena(s), 2 dezena(s) e 4 unidade(s)


**Exercício 20.**

* Faça um Programa para leitura de três notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e presentar:

* A mensagem "Aprovado", se a média for maior ou igual a 7, com a respectiva média alcançada;
* A mensagem "Reprovado", se a média for menor do que 7, com a respectiva média alcançada;
* A mensagem "Aprovado com Distinção", se a média for igual a 10.

In [ ]:
print ('Checagem de Aprovação Trimestral.')
print (' '*43)
aluno = input('Por favor, insira o nome do aluno: ')
print (' '*43)
nota1 = float( input('Informe a 1ª nota: '))
nota2 = float( input('Informe a 2ª nota: '))
nota3 = float( input('Informe a 3ª nota: '))

if nota1 < 0 and nota2 < 0 and nota3 > 0 or nota1 <= 10 and nota2 <=10 and nota3 <= 10:
  media = (nota1+nota2+nota3)/3
  if media >= 7 and media < 9.75:
    print (' '*43)
    print (f'O aluno {aluno} foi Aprovado. Sua média trimestral foi {media:.2f}.')
  if media == 10:
    print (' '*43)
    print (f'O aluno {aluno} foi Aprovado com Distinção. Sua média trimestral foi {media:.2f}.')
  if media < 6.75:
    print (' '*43)
    print (f'O aluno {aluno} infelizmente foi Reprovado. Sua média trimestral foi {media:.2f}.')

else:
  print (' '*43)
  print ('Valor de nota invalido.')


Checagem de Aprovação Trimestral.
                                           
Por favor, insira o nome do aluno: Luiz
                                           
Informe a 1ª nota: 6.5
Informe a 2ª nota: 6.5
Informe a 3ª nota: 10
                                           
O aluno Luiz foi Aprovado. Sua média trimestral foi 7.67.


**Exercício 21.**

* Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao usuário a valor do saque e depois informar quantas notas de cada valor serão fornecidas. 

  As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a quantidade de notas existentes na máquina.

  a) Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de 100, uma nota de 50, uma nota de 5 e uma nota de 1;

  b) Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de 100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.

In [ ]:
print ('Bem-vindo ao Caixa Eletrônico do Banco LGO.')
print (' '*45)
saque = float(input('Favor informar o valor do saque desejado: '))
if saque >=10 and saque <= 600:  
  notas_de_100 = notas_de_50 = notas_de_10 = notas_de_5 = notas_de_1 = 0
# Notas de 100
  notas_de_100 , saque = divmod(saque,100)
# Notas de 50
  notas_de_50, saque = divmod(saque,50)
# Notas de 10
  notas_de_10, saque = divmod(saque,10)
# Notas de 5
  notas_de_5, saque = divmod(saque,5)
# Notas de 1
  notas_de_1, saque = divmod(saque,1)
# Impressao
  print (' '*45)
  print (f'{int(notas_de_100)} nota(s) de R$100') 
  print (f'{int(notas_de_50) } nota(s) de R$50') 
  print (f'{int(notas_de_10) } nota(s) de R$10') 
  print (f'{int(notas_de_5)  } nota(s) de R$5')
  print (f'{int(notas_de_1)  } nota(s) de R$1')

else:
  print ('Valor Indisponível.')

Bem-vindo ao Caixa Eletrônico do Banco LGO.
                                             
Favor informar o valor do saque desejado: 548
                                             
5 nota(s) de R$100
0 nota(s) de R$50
4 nota(s) de R$10
1 nota(s) de R$5
3 nota(s) de R$1


**Exercício 22.**

* Faça um Programa que peça um número inteiro e determine se ele é par ou impar. Dica: utilize o operador módulo (resto da divisão).

In [ ]:
numero = int( input('Digite um número inteiro: '))
if (numero%2) == 0:
  print("Par.")
else:
  print("Ímpar.")

Digite um número inteiro: 5487
Ímpar.


**Exercício 23.**
* Faça um Programa que peça um número e informe se o número é inteiro ou decimal. Dica: utilize uma função de arredondamento.

In [ ]:
numero = input('Digite um número: ')
try:
    int(numero)
    inteiro = True
    print ('Número Inteiro.')
except ValueError:
    inteiro = False
    print('Número Decimal.')

Digite um número: 1.1254
Número Decimal.


**Exercício 24.**
* Faça um Programa que leia 2 números e em seguida pergunte ao usuário qual operação ele deseja realizar. 

  O resultado da operação deve ser acompanhado de uma frase que diga se o número é:
  * par ou ímpar;
  * positivo ou negativo;
  * inteiro ou decimal.

In [ ]:
numero1 = eval( input('Digite um número: '))
numero2 = eval( input('Digite outro número: '))

operacao = input('Qual operação você deseja realizar? (+ , - , / , * , **) ')
while operacao != '+' and operacao != '-' and operacao != '/' and operacao != '*' and operacao != '**':
  print ('Operador Inválido. Tente novamente.')
  operacao = input('Qual operação você deseja realizar? (+ , - , / , * , **) ')

#Operador +
if operacao == '+':
  calculo = numero1 + numero2
  if (calculo%2) == 0:
    print(f'O resultado {calculo} é Par.')
  else:
    print(f'O resultado {calculo} é Ímpar.')
  if calculo > 0:
    print('Positivo.')
  else:
    print('Negativo.')
  if calculo == int(calculo):
    print('Número Inteiro.')
  else:
    print('Número Decimal.')  

#Operador -
elif operacao == '-':
  calculo = numero1 - numero2
  print (calculo)
  if (calculo%2) == 0:
    print(f'O resultado {calculo} é Par.')
  else:
    print(f'O resultado {calculo} é Ímpar.')
  if calculo > 0:
    print('Positivo.')
  else:
    print('Negativo.')
  if calculo == int(calculo):
    print('Número Inteiro.')
  else:
    print('Número Decimal.') 

#Operador /
elif operacao == '/':
  calculo = numero1 / numero2
  print (calculo)
  if (calculo%2) == 0:
    print(f'O resultado {calculo} é Par.')
  else:
    print(f'O resultado {calculo} é Ímpar.')
  if calculo > 0:
    print('Positivo.')
  else:
    print('Negativo.')
  if calculo == int(calculo):
    print('Número Inteiro.')
  else:
    print('Número Decimal.') 

#Operador *
elif operacao == '*':
  calculo = numero1 * numero2
  print (calculo)
  if (calculo%2) == 0:
    print(f'O resultado {calculo} é Par.')
  else:
    print(f'O resultado {calculo} é Ímpar.')
  if calculo > 0:
    print('Positivo.')
  else:
    print('Negativo.')
  if calculo == int(calculo):
    print('Número Inteiro.')
  else:
    print('Número Decimal.') 

#Operador **
else: 
  if operacao == '**':
    calculo = numero1 ** numero2
    print (calculo)
  if (calculo%2) == 0:
    print(f'O resultado {calculo} é Par.')
  else:
    print(f'O resultado {calculo} é Ímpar.')
  if calculo > 0:
    print('Positivo.')
  else:
    print('Negativo.')
  if calculo == int(calculo):
    print('Número Inteiro.')
  else:
    print('Número Decimal.') 

Digite um número: 14
Digite outro número: 2
Qual operação você deseja realizar? (+ , - , / , * , **) **
196
O resultado 196 é Par.
Positivo.
Número Inteiro.


**Exercício 25.**

* Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As perguntas são:

  "Telefonou para a vítima?"

  "Esteve no local do crime?"
  
  "Mora perto da vítima?"

  "Devia para a vítima?"

  "Já trabalhou com a vítima?" 
  
  O programa deve no final emitir uma classificação sobre a participação da pessoa no crime. 
  
    * Se a pessoa responder positivamente a 2 questões ela deve ser classificada como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário, ele será classificado como "Inocente".

In [ ]:
print ('Delegacia de Policia do Estado de SP.')
print (' ')
print ('Para melhor esclarecimento das investigações responda "SIM" ou "NÃO" para as seguintes pergutnas:')
print (' ')
resposta = 0

resposta1 = str(input('Telefonou para a vítima? '))
resposta1 = resposta1.lower()
while resposta1 != 'sim' and resposta1 != 'não':
  print ('Resposta incorreta. Por favor, responda SIM ou NÃO.')
  resposta1 = str(input('Telefonou para a vítima? '))
  resposta1 = resposta1.lower()
else:
  if resposta1 == 'sim':  
    resposta+= 1
  else:
    resposta+= 0

resposta2 = str(input('Esteve no local do crime? '))
resposta2 = resposta2.lower()
while resposta2 != 'sim' and resposta2 != 'não':
  print ('Resposta incorreta. Por favor, responda SIM ou NÃO.')
  resposta2 = str(input('Esteve no local do crime? '))
  resposta2 = resposta2.lower()
else:
  if resposta2 == 'sim':  
    resposta+= 1
  else:
    resposta+= 0

resposta3 = str(input('Mora perto da vítima? '))
resposta3 = resposta3.lower()
while resposta3 != 'sim' and resposta3 != 'não':
  print ('Resposta incorreta. Por favor, responda SIM ou NÃO.')
  resposta3 = str(input('Mora perto da vítima? '))
  resposta3 = resposta3.lower()
else:
  if resposta3 == 'sim':  
    resposta+= 1
  else:
    resposta+= 0

resposta4 = str(input('Devia para a vítima? '))
resposta4 = resposta4.lower()
while resposta4 != 'sim' and resposta4 != 'não':
  print ('Resposta incorreta. Por favor, responda SIM ou NÃO.')
  resposta4 = str(input('Devia para a vítima? '))
  resposta4 = resposta4.lower()
else:
  if resposta4 == 'sim':  
    resposta+= 1
  else:
    resposta+= 0

resposta5 = str(input('Já trabalhou com a vítima? '))
resposta5 = resposta5.lower()
while resposta5 != 'sim' and resposta5 != 'não':
  print ('Resposta incorreta. Por favor, responda SIM ou NÃO.')
  resposta5 = str(input('Já trabalhou com a vítima? '))
  resposta5 = resposta5.lower()
else:
  if resposta5 == 'sim':  
    resposta+= 1
  else:
    resposta+= 0


#Conclusão
if resposta == 2:
  print ('Suspeito.')
elif resposta >= 3 and resposta <= 4:
  print ('Cúmplice.')
elif resposta == 5:
  print ('Assassino.')
else:
  print (' ')
  print ('Inocente.')

Delegacia de Policia do Estado de SP.
 
Para o esclarecimento das investigações responda "SIM" ou "NÃO" as seguintes pergutnas:
 
Telefonou para a vítima? Sim
Esteve no local do crime? Sim
Mora perto da vítima? Sim
Devia para a vítima? Sim
Já trabalhou com a vítima? Sim
Assassino.


**Exercício 26.**

Um posto está vendendo combustíveis com a seguinte tabela de descontos:

Álcool:
* até 20 litros, desconto de 3% por litro
* acima de 20 litros, desconto de 5% por litro

Gasolina:
* até 20 litros, desconto de 4% por litro
* acima de 20 litros, desconto de 6% por litro 

Escreva um algoritmo que leia: 

  * o número de litros vendidos,
  * o tipo de combustível (codificado da seguinte forma: A-álcool, G-gasolina)
  * calcule e imprima o valor a ser pago pelo cliente
    obs: sabendo-se que o preço do litro da gasolina é R$ 2,50 o preço do litro do álcool é R$ 1,90.

In [ ]:
print ('Bem vindo ao Posto de Gasolina LGO.')
print (''*4)

a_ou_g = input('Álcool ou Gasolina? (Digite a inicial da opção desejada): ')
a_ou_g = a_ou_g.lower()
litros = float( input('Quantos litros você deseja abastescer? '))

#Álcool
if a_ou_g == 'a' and litros <= 20:
  desconto = litros * 3/100
  preco_alcool = 1.9
  print (f'O total ficou R$ {(preco_alcool * litros) - desconto:.2f} por {litros} litros de álcool.')
  print (f'Desconto de: R${desconto:.2f} .')
else:
  if a_ou_g == 'a':
    desconto = litros * 5/100
    preco_alcool = 1.9
    print (f'O total ficou R$ {(preco_alcool * litros) - desconto:.2f} por {litros} litros de álcool.')
    print (f'Desconto de: R${desconto:.2f} .')

#Gasolina
if a_ou_g == 'g' and litros <= 20:
  desconto = litros * 4/100
  preco_gasolina = 2.5
  print (f'O total ficou R$ {(preco_gasolina * litros) - desconto:.2f} por {litros} litros de gasolina.')
  print (f'Desconto de: R${desconto:.2f} .')
else:
  if a_ou_g == 'g':
    desconto = litros * 6/100
    preco_gasolina = 2.5
    print (f'O total ficou R$ {(preco_gasolina * litros) - desconto:.2f} por {litros} litros de gasolina.')
    print (f'Desconto de: R${desconto:.2f} .')


Bem vindo ao Posto de Gasolina LGO.

Álcool ou Gasolina? (Digite a inicial da opção desejada): A
Quantos litros você deseja abastescer? 20
O total ficou R$ 37.40 por 20.0 litros de álcool.
Desconto de: R$0.60 .


**Exercício 27.**

Uma fruteira está vendendo frutas com a seguinte tabela de preços:

                       Até 5 Kg              Acima de 5 Kg
    Morango         R$ 2,50 por Kg          R$ 2,20 por Kg
    Maçã            R$ 1,80 por Kg          R$ 1,50 por Kg

Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar R$ 25,00, receberá ainda um desconto de 10% sobre este total.

Escreva um algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de maças adquiridas e escreva o valor a ser pago pelo cliente.

In [ ]:
print ('Bem-vindo ao Super Mercado LGO!')
print (' '*20)

morango_kg = float( input('Quantos kilos de morango você deseja comprar? '))
maca_kg = float( input('Quantos kilos de maça você deseja comprar? '))

#Morango
if morango_kg <= 5:
  preco_morango = 2.5
  total_morango = morango_kg * preco_morango
  print (' '*10)
  print (f'R$ {total_morango:.2f} para {morango_kg:.2f} kg(s) de morango.')

else:
  preco_morango = 2.2
  total_morango = morango_kg * preco_morango
  print (' '*10)
  print (f'R$ {total_morango:.2f} para {morango_kg:.2f} kg(s) de morango.')

#Maçã
if maca_kg <= 5:
  preco_maca = 1.8
  total_maca = maca_kg * preco_maca
  print (' '*10)
  print (f'R$ {total_maca:.2f} para {maca_kg:.2f} kg(s) de maçã.')

else:
  preco_maca = 1.5
  total_maca = maca_kg * preco_maca
  print (' '*10)
  print (f'R$ {total_maca:.2f} para {maca_kg:.2f} kg(s) de maçã.')


#Total:
frutas_kg = morango_kg + maca_kg
total_frutas = total_morango + total_maca

if frutas_kg > 8 or total_frutas > 25:
  desconto = total_frutas * 10/100
  print (' '*10)
  print (f'Valor da compra: R$ {total_morango + total_maca:.2f}.')
  print (f'Você ganhou um desconto de R${desconto:.2f}. Valor da compra R${total_frutas - desconto:.2f} .')

else:
  print (' '*10)
  print (f'Valor da compra: R$ {total_morango + total_maca:.2f} .')

Bem-vindo ao Super Mercado LGO!
                    
Quantos kilos de morango você deseja comprar? 6
Quantos kilos de maça você deseja comprar? 8
          
R$ 13.20 para 6.00 kg(s) de morango.
          
R$ 12.00 para 8.00 kg(s) de maçã.
          
Valor da compra: R$ 25.20.
Você ganhou um desconto de R$2.52. Valor da compra R$22.68 para 14.0 kgs de frutas.


**Exercício 28.**

O Hipermercado Tabajara está com uma promoção de carnes que é imperdível. Confira:

                       Até 5 Kg              Acima de 5 Kg
    File Duplo      R$ 4,90 por Kg          R$ 5,80 por Kg
    Alcatra         R$ 5,90 por Kg          R$ 6,80 por Kg
    Picanha         R$ 6,90 por Kg          R$ 7,80 por Kg


Para atender a todos os clientes, cada cliente poderá levar apenas um dos tipos de carne da promoção, porém não há limites para a quantidade de carne por cliente. 

Se compra for feita no cartão Tabajara o cliente receberá ainda um desconto de 5% sobre o total da compra. 

* Escreva um programa que peça o tipo e a quantidade de carne comprada pelo usuário e gere um cupom fiscal, contendo as informações da compra: 
  * Tipo e quantidade de carne, 
  * Preço total, 
  * Tipo de pagamento, 
  * Valor do desconto e valor a pagar.

In [ ]:
print ('Bem-vindo ao Super Mercado Tabajara!')
print ('Estamos com uma promoção de carnes imperdível.')
print (' '*10)
print ('Opções: File Duplo / Alcatra / Picanha.')
print (' '*10)
carne_escolhida = input ('Digite a carne que você deseja comprar: ')
carne_escolhida = carne_escolhida.lower()
while carne_escolhida != 'file duplo' and carne_escolhida != 'alcatra' and carne_escolhida != 'picanha':
  print ('Valor Inválido. Tente novamente')
  carne_escolhida = input ('Digite a carne que você deseja comprar: ')
  carne_escolhida = carne_escolhida.lower()

### File Duplo ###
if carne_escolhida == 'file duplo':
  print (''*10)
  kg_fileduplo = float( input('Quantos kilos de File Duplo você gostaria de comprar? '))
  if kg_fileduplo <= 5:
    preco_fileduplo = 4.9
    total_fileduplo = kg_fileduplo * preco_fileduplo

#Cartao Tabajara kg por R$ 4.90
    client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
    client_confirm = client_confirm.lower()
    while client_confirm != 's' and client_confirm != 'n':
      print ('Valor Inválido. Tente novamente')
      client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
      client_confirm = client_confirm.lower()
    else:
      if client_confirm == 's':
        desconto = total_fileduplo * 5/100
        print (''*10)
        print (f'Valor da compra R$ {total_fileduplo - desconto:.2f} para {kg_fileduplo:.2f} kg(s) de File Duplo.')
        print (f'Desconto Cliente Tabajara: (-) R$ {desconto:.2f} .')
      else:
        print (''*10)
        print (f'Valor da compra R$ {total_fileduplo:.2f} para {kg_fileduplo:.2f} kg(s) de File Duplo.')
  else:
    preco_fileduplo = 5.8
    total_fileduplo = kg_fileduplo * preco_fileduplo

#Cartao Tabajara kg por R$ 5.80
    client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
    client_confirm = client_confirm.lower()
    while client_confirm != 's' and client_confirm != 'n':
      print ('Valor Inválido. Tente novamente')
      client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
      client_confirm = client_confirm.lower()
    else:
      if client_confirm == 's':
        desconto = total_fileduplo * 5/100
        print (''*10)
        print (f'Valor da compra R$ {total_fileduplo - desconto:.2f} para {kg_fileduplo:.2f} kg(s) de File Duplo.')
        print (f'Desconto Cliente Tabajara: (-) R$ {desconto:.2f} .')
      else:
        print (''*10)
        print (f'Valor da compra R$ {total_fileduplo:.2f} para {kg_fileduplo:.2f} kg(s) de File Duplo.')


### Alcatra ###
elif carne_escolhida == 'alcatra':
  print (''*10)
  kg_alcatra = float( input('Quantos kilos de Alcatra você gostaria de comprar? '))
  if kg_alcatra <= 5:
    preco_alcatra = 5.9
    total_alcatra = kg_alcatra * preco_alcatra

#Cartao Tabajara kg por R$ 5.90
    client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
    client_confirm = client_confirm.lower()
    while client_confirm != 's' and client_confirm != 'n':
      print ('Valor Inválido. Tente novamente')
      client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
      client_confirm = client_confirm.lower()
    else:
      if client_confirm == 's':
        desconto = total_alcatra * 5/100
        print (''*10)
        print (f'Valor da compra R$ {total_alcatra - desconto:.2f} para {kg_alcatra:.2f} kg(s) de Alcatra.')
        print (f'Desconto Cliente Tabajara: (-) R$ {desconto:.2f} .')
      else:
        print (''*10)
        print (f'Valor da compra R$ {total_alcatra:.2f} para {kg_alcatra:.2f} kg(s) de Alcatra.')
  else:
    preco_alcatra = 6.8
    total_alcatra = kg_alcatra * preco_alcatra

#Cartao Tabajara kg por R$ 6.80
    client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
    client_confirm = client_confirm.lower()
    while client_confirm != 's' and client_confirm != 'n':
      print ('Valor Inválido. Tente novamente')
      client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
      client_confirm = client_confirm.lower()
    else:
      if client_confirm == 's':
        desconto = total_alcatra * 5/100
        print (''*10)
        print (f'Valor da compra R$ {total_alcatra - desconto:.2f} para {kg_alcatra:.2f} kg(s) de Alcatra.')
        print (f'Desconto Cliente Tabajara: (-) R$ {desconto:.2f} .')
      else:
        print (''*10)
        print (f'Valor da compra R$ {total_alcatra:.2f} para {kg_alcatra:.2f} kg(s) de Alcatra.')


### Picanha ###
if carne_escolhida == 'picanha':
  print (''*10)
  kg_picanha = float( input('Quantos kilos de Picanha você gostaria de comprar? '))
  if kg_picanha <= 5:
    preco_picanha = 6.9
    total_picanha = kg_picanha * preco_picanha

#Cartao Tabajara kg por R$ 6.90
    client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
    client_confirm = client_confirm.lower()
    while client_confirm != 's' and client_confirm != 'n':
      print ('Valor Inválido. Tente novamente')
      client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
      client_confirm = client_confirm.lower()
    else:
      if client_confirm == 's':
        desconto = total_picanha * 5/100
        print (''*10)
        print (f'Valor da compra R$ {total_picanha - desconto:.2f} para {kg_picanha:.2f} kg(s) de Picanha.')
        print (f'Desconto Cliente Tabajara: (-) R$ {desconto:.2f} .')
      else:
        print (''*10)
        print (f'Valor da compra R$ {total_picanha:.2f} para {kg_picanha:.2f} kg(s) de Picanha.')
  else:
    preco_picanha = 7.8
    total_picanha = kg_picanha * preco_picanha

#Cartao Tabajara kg por R$ 7.80
    client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
    client_confirm = client_confirm.lower()
    while client_confirm != 's' and client_confirm != 'n':
      print ('Valor Inválido. Tente novamente')
      client_confirm = input ('Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): ')
      client_confirm = client_confirm.lower()
    else:
      if client_confirm == 's':
        desconto = total_picanha * 5/100
        print (''*10)
        print (f'Valor da compra R$ {total_picanha - desconto:.2f} para {kg_picanha:.2f} kg(s) de Picanha.')
        print (f'Desconto Cliente Tabajara: (-) R$ {desconto:.2f} .')
      else:
        print (''*10)
        print (f'Valor da compra R$ {total_picanha:.2f} para {kg_picanha:.2f} kg(s) de Picanha.')

print (' '*20)
print ('O Super Mercado Tabajara Agradece. Volte sempre!')

Bem-vindo ao Super Mercado Tabajara!
Estamos com uma promoção de carnes imperdível.
          
Opções: File Duplo / Alcatra / Picanha.
          
Digite a carne que você deseja comprar: aLCATRA

Quantos kilos de Alcatra você gostaria de comprar? 7.5
Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): nAO
Valor Inválido. Tente novamente
Pagamento com Cartão Tabajara? (S para SIM / N para NÂO): N

Valor da compra R$ 51.00 para 7.50 kg(s) de Alcatra.
                    
O Super Mercado Tabajara Agradece. Volte sempre!


#[Estrutura de Repetição](https://wiki.python.org.br/EstruturaDeRepeticao)
* Em andamento...

**Exercício 1.**
* Faça um programa que peça uma nota, entre zero e dez. 

  Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.


In [ ]:
while True:
  try:
    numero = int( input('Informe um número de 1 a 10: '))
  except ValueError:
    print ('Deve ser fornecido um valor inteiro.')
  else:
    if 0 <= numero <= 10:
      print (f'O número informado foi: {numero}.') 
      break
    else:
      print ('O número deve estar entre 0 e 10.')

Informe um número de 1 a 10: 5
O número informado foi: 5.


**Exercício 2.**

* Faça um programa que leia um nome de usuário e a sua senha e não aceite a senha igual ao nome do usuário, mostrando uma mensagem de erro e voltando a pedir as informações.


In [ ]:
ok = False
while not ok:
  try:
    login = input('Informe um login para acesso : ')
    senha = input('Informe uma senha para acesso: ')
  except ValueError:
    print (' ')
  if login == senha:
      print (f'O login e senha não podem ser iguais.') 
  else:
      print ('Cadastro realizado com sucesso.')
      break

Informe um login para acesso : luizGouveia
Informe uma senha para acesso: luizGouveia
O login e senha não podem ser iguais.
Informe um login para acesso : lgo1234
Informe uma senha para acesso: lgo1234
O login e senha não podem ser iguais.
Informe um login para acesso : 32578LL
Informe uma senha para acesso: 32578LL
O login e senha não podem ser iguais.
Informe um login para acesso : lgou
Informe uma senha para acesso: 9999
Cadastro realizado com sucesso.


**Exercício 3.**

* Faça um programa que leia e valide as seguintes informações:
  
  Nome: maior que 3 caracteres;

  Idade: entre 0 e 150;

  Salário: maior que zero;

  Sexo: 'f' ou 'm';

  Estado Civil: 's', 'c', 'v', 'd';


In [ ]:
#Nome
nome = str( input('Informe seu nome : '))
while len(nome) <= 2:
  print (f'O nome deve conter mais que 3 caracteres.')
  nome = str( input('Informe seu nome : '))
#Idade
idade = int(input('Informe sua idade: '))
while idade < 0 or idade > 150:  
  print (f'A idade deve ser entre 0 a 150 anos.')
  idade = int(input('Informe sua idade: '))
#Salario   
salario = float(input('Informe seu salário: '))
while salario < 0:
  print (f'O salário deve ser maior que 0.')  
  salario = float(input('Informe seu salário: '))
#Sexo
sexo = str(input('Informe seu sexo ("f" para feminino / "m" para masculino): '))
sexo = sexo.lower()
while sexo != 'f' and sexo != 'm':  
  print (f'O sexo informado deve ser "F" para feminino ou "M" para masculino.') 
  sexo = str(input('Informe seu sexo ("F" para feminino / "M" para masculino): '))
  sexo = sexo.lower()
#EstadoCivil  
e_civil = str(input('Informe seu estado civil ("S" para solteiro(a) , "C" para Casado(a), "V" para viúvo(a) ou "D" para divorciado(a): '))
e_civil = e_civil.lower()
while e_civil != 's' and e_civil != 'c' and e_civil != "v" and e_civil != "d":
  print (f'Valor inválido.')
  e_civil = str(input('Informe seu estado civil ("S" para solteiro(a) , "C" para Casado(a), "V" para viúvo(a) ou "D" para divorciado(a): '))
  e_civil = e_civil.lower()
#Break
else:
  print ('Cadastro realizado com sucesso.')

Informe seu nome : Lg
O nome deve conter mais que 3 caracteres.
Informe seu nome : Luiz
Informe sua idade: -87
A idade deve ser entre 0 a 150 anos.
Informe sua idade: -856
A idade deve ser entre 0 a 150 anos.
Informe sua idade: 32
Informe seu salário: -8754
O salário deve ser maior que 0.
Informe seu salário: 9000
Informe seu sexo ("f" para feminino / "m" para masculino): G
O sexo informado deve ser "f" para feminino ou "m" para masculino.
Informe seu sexo ("f" para feminino / "m" para masculino): M
Informe seu estado civil ("S" para solteiro(a) , "C" para Casado(a), "V" para viúvo(a) ou "D" para divorciado(a): F
Valor inválido.
Informe seu estado civil ("S" para solteiro(a) , "C" para Casado(a), "V" para viúvo(a) ou "D" para divorciado(a): D
Cadastro realizado com sucesso.


**Exercício 4.**
* Supondo que a população de um país A seja da ordem de 80000 habitantes com uma taxa anual de crescimento de 3% e que a população de B seja 200000 habitantes com uma taxa de crescimento de 1.5%. Faça um programa que calcule e escreva o número de anos necessários para que a população do país A ultrapasse ou iguale a população do país B, mantidas as taxas de crescimento.


In [ ]:
populacao_a = 80_000
populacao_b = 200_000
taxa_crescimento_a = 1.03
taxa_crescimento_b = 1.015
ano = 0

while populacao_a < populacao_b:
  print (f'##### Populações no ano {ano}.')
  print (' '*10) 
  print (f'População de A: {populacao_a} .')
  print (f'População de B: {populacao_b} .')
  ano += 1
  populacao_a = int(populacao_a * taxa_crescimento_a)
  populacao_b = int (populacao_b * taxa_crescimento_b)
# outra forma de fazer a mesma conta seria:
# populacao_a *= taxa_crescimento_a
#outra forma de converter em inteiro seria
# populacao_a = int (populacao_a)
print (' '*2)
print (f'{ano} anos para a população do país A se igualar ou ultrapassar a população do país B.')
print (f'População de A: {populacao_a} .')
print (f'População de B: {populacao_b} .')

##### Populações no ano 0.
          
População de A: 80000 .
População de B: 200000 .
##### Populações no ano 1.
          
População de A: 82400 .
População de B: 202999 .
##### Populações no ano 2.
          
População de A: 84872 .
População de B: 206043 .
##### Populações no ano 3.
          
População de A: 87418 .
População de B: 209133 .
##### Populações no ano 4.
          
População de A: 90040 .
População de B: 212269 .
##### Populações no ano 5.
          
População de A: 92741 .
População de B: 215453 .
##### Populações no ano 6.
          
População de A: 95523 .
População de B: 218684 .
##### Populações no ano 7.
          
População de A: 98388 .
População de B: 221964 .
##### Populações no ano 8.
          
População de A: 101339 .
População de B: 225293 .
##### Populações no ano 9.
          
População de A: 104379 .
População de B: 228672 .
##### Populações no ano 10.
          
População de A: 107510 .
População de B: 232102 .
##### Populações no ano 11.
          


**Exercício 5.**

* Altere o programa anterior permitindo ao usuário informar as populações e as taxas de crescimento iniciais. Valide a entrada e permita repetir a operação.


In [ ]:
print ('Descubra em quantos anos a população do país A se igualará ou ultrapassará o país B.')
print (' '*10)

##### VALIDADOR DE NÚMERO INTEIRO #####

### POPULAÇÃO A ###
def leiaInt (mensagem):
  ok = False
  valor = 0
  while True:
    populacao_a = str(input(mensagem))
    if populacao_a.isnumeric():
      valor = int(populacao_a)
      ok = True
    else:
      print ('Valor Inválido. Tente novamente.')
    if ok:
      break
  return valor

### POPULAÇÃO B ###
def leiaInt (mensagem):
  ok = False
  valor = 0
  while True:
    populacao_b = str(input(mensagem))
    if populacao_b.isnumeric():
      valor = int(populacao_b)
      ok = True
    else:
      print ('Valor Inválido. Tente novamente.')
    if ok:
      break
  return valor

##########################################
### INPUT POPULAÇÃO ###
populacao_a = leiaInt('Digite o número da população A: ')
populacao_b = leiaInt('Digite o número da população B: ')

##########################################
### INPUT TAXA DE CRESCIMENTO ###

# VALIDADOR DE PONTO FLUTUANTE #
### TAXA DE CRESCIMENTO PAÍS A ###
ok = False
while not ok:
  try:
    taxa_crescimento_a = float( input('Digite a Taxa de Crescimento do País A: '))
    ok = True
  except ValueError:
    print ('Valor Inválido.')

### TAXA DE CRESCIMENTO PAÍS B ###
ok = False
while not ok:
  try:
    taxa_crescimento_b = float( input('Digite a Taxa de Crescimento do País B: '))
    ok = True
  except ValueError:
    print ('Valor Inválido.')
##########################################

ano = 0

while populacao_a < populacao_b:
  print (f'##### Populações no ano {ano}.')
  print (' '*10) 
  print (f'População de A: {populacao_a} .')
  print (f'População de B: {populacao_b} .')
  ano += 1
  populacao_a = int(populacao_a * taxa_crescimento_a)
  populacao_b = int (populacao_b * taxa_crescimento_b)
# outra forma de fazer a mesma conta seria:
# populacao_a *= taxa_crescimento_a
#outra forma de converter em inteiro seria
# populacao_a = int (populacao_a)
print (' '*2)
print (f'{ano} anos para a população do país A se igualar ou ultrapassar a população do país B.')
print (f'População de A: {populacao_a} .')
print (f'População de B: {populacao_b} .')

Descubra em quantos anos a população do país A se igualará ou ultrapassará o país B.
          
Digite o número da população A: 5siuhdiuahde
Valor Inválido. Tente novamente.
Digite o número da população A: 10000
Digite o número da população B: kjdabdbaieda
Valor Inválido. Tente novamente.
Digite o número da população B: 50000
Digite a Taxa de Crescimento do País A: daiuhdeuda
Valor Inválido.
Digite a Taxa de Crescimento do País A: dbaibdeda
Valor Inválido.
Digite a Taxa de Crescimento do País A: 1.5
Digite a Taxa de Crescimento do País B: daiudheuida
Valor Inválido.
Digite a Taxa de Crescimento do País B: 1.3
##### Populações no ano 0.
          
População de A: 10000 .
População de B: 50000 .
##### Populações no ano 1.
          
População de A: 15000 .
População de B: 65000 .
##### Populações no ano 2.
          
População de A: 22500 .
População de B: 84500 .
##### Populações no ano 3.
          
População de A: 33750 .
População de B: 109850 .
##### Populações no ano 4.
          


**Exercício 6.**
* Faça um programa que imprima na tela os números de 1 a 20, um abaixo do outro. Depois modifique o programa para que ele mostre os números um ao lado do outro.

In [ ]:
i = 0
lista = list (range(21))

while i < len(lista):
  print (lista[i])
  i += 1

print (lista)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]


**Exercício 7.**  
* Faça um programa que leia 5 números e informe o maior número.

In [ ]:
ok = False
while not ok:
  try:
    maximo = float(input('Digite um número: '))
    ok = True
  except ValueError:
    print ('Valor Inválido. Tente novamente.')

for _ in range(4):
  maximo = max(maximo, float(input('Digite um número: ')))
  print (f'Número maximo encontrado até agora é {maximo}')

Digite um número: suhauhes
Valor Inválido. Tente novamente.
Digite um número: 234
Digite um número: 1234
Número maximo encontrado até agora é 1234.0
Digite um número: 23
Número maximo encontrado até agora é 1234.0
Digite um número: 234
Número maximo encontrado até agora é 1234.0
Digite um número: 456
Número maximo encontrado até agora é 1234.0
Digite um número: 5000
Número maximo encontrado até agora é 5000.0
Digite um número: 5001
Número maximo encontrado até agora é 5001.0


**Exercício 8.**

* Faça um programa que leia 5 números e informe a soma e a média dos números.

In [ ]:
ok = False
while not ok:
  try:
    numero = float(input('Digite um número: '))
    ok = True
  except ValueError:
    print ('Valor Inválido. Tente novamente.')

for i in range(4):
  numero = (numero + float(input('Digite mais um número: ')))  
  i += 2
  media = numero / i
  print (' '*2)
  print (f'Até agora a soma dos números é {numero:.2f}')
  print (f'A média dos {i} números somados é {media:.2f}')
  print (' '*2)

Digite um número: 10
Digite mais um número: 10
  
Até agora a soma dos números é 20.00
A média dos 2 números somados é 10.00
  
Digite mais um número: 10
  
Até agora a soma dos números é 30.00
A média dos 3 números somados é 10.00
  
Digite mais um número: 10
  
Até agora a soma dos números é 40.00
A média dos 4 números somados é 10.00
  
Digite mais um número: 10
  
Até agora a soma dos números é 50.00
A média dos 5 números somados é 10.00
  


**Exercício 9**
* Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.

In [ ]:
start, end = 1, 50
for num in range (start, end +1):
  if num % 2 != 0:
    print (num, end = " ")

1 3 5 7 9 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

**Exercício 10.**
* Faça um programa que receba dois números inteiros e gere os números inteiros que estão no intervalo compreendido por eles.

In [ ]:
### VALIDADOR NUMERO INTEIRO - START ###
def leiaInt (mensagem):
  ok = False
  valor = 0
  while True:
    start = str(input(mensagem))
    if start.isnumeric():
      valor = int(start)
      ok = True
    else:
      print ('O número não é inteiro. Tente novamente.')
    if ok:
      break
  return valor

# INPUT START #
start = leiaInt('Digite um número inteiro: ')

### VALIDADOR NUMERO INTEIRO - END ###
def leiaInt (mensagem):
  ok = False
  valor = 0
  while True:
    end = str(input(mensagem))
    if end.isnumeric():
      valor = int(end)
      ok = True
    else:
      print ('O número não é inteiro. Tente novamente.')
    if ok:
      break
  return valor

# INPUT END #
end = leiaInt('Digite outro número inteiro: ')

# IMPRESSÃO INTERVALO #
print (' '*2)
print (f'Os números dentro do intervalo entre {start} e {end} são:')
for i in range(start +1,end):
  print (i, end =' ')
  
# OU...
#for num in range (start, end):
 # if num > start and num < end:
  #  print (num , end = " ")

Digite um número inteiro: 1
Digite outro número inteiro: 10
  
Os números dentro do intervalo entre 1 e 10 são:
2 3 4 5 6 7 8 9 

**Exercício 11.**
* Altere o programa anterior para mostrar no final a soma dos números.

In [ ]:
### VALIDADOR NUMERO INTEIRO - START ###
def leiaInt (mensagem):
  ok = False
  valor = 0
  while True:
    start = str(input(mensagem))
    if start.isnumeric():
      valor = int(start)
      ok = True
    else:
      print ('O número não é inteiro. Tente novamente.')
    if ok:
      break
  return valor

# INPUT START #
start = leiaInt('Digite um número inteiro: ')

### VALIDADOR NUMERO INTEIRO - END ###
def leiaInt (mensagem):
  ok = False
  valor = 0
  while True:
    end = str(input(mensagem))
    if end.isnumeric():
      valor = int(end)
      ok = True
    else:
      print ('O número não é inteiro. Tente novamente.')
    if ok:
      break
  return valor

# INPUT END #
end = leiaInt('Digite outro número inteiro: ')

print (''*2)

# IMPRESSÃO INTERVALO ENTRE START - END (CONTANDO OS INPUTS)#
print (f'Vamos analisar a extensão de {start} a {end}:')
for i in range(start,end +1):
  print (i, end =' ')
print (end='\n')
print (f'A soma de {start} a {end} é: {sum(range(start,end+1))}')

print (''*2)

# IMPRESSÃO INTERVALO ENTRE START - END (S/ CONTAR OS INPUTS)
print (f'Agora analisaremos o intervalo entre {start} e {end}:')
for i in range(start+1,end):
  print (i, end =' ')
print (end='\n')
print (f'A soma do intervalo, indo de {start+1} até {end-1}, é: {sum(range(start+1,end))}')

Digite um número inteiro: 1
Digite outro número inteiro: 10

Vamos analisar a extensão de 1 a 10:
1 2 3 4 5 6 7 8 9 10 
A soma de 1 a 10 é: 55

Agora analisaremos o intervalo entre 1 e 10:
2 3 4 5 6 7 8 9 
A soma do intervalo, indo de 2 até 9, é: 44


**Exercício 12.**

* Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver a tabuada. A saída deve ser conforme o exemplo abaixo:

      Tabuada de 5:

      5 X 1 = 5
      5 X 2 = 10
      ...
      5 X 10 = 50

In [ ]:
ok = False
while not ok:
  try:
    numero = int(input('Digite um número inteiro de 1 a 10: '))
    ok = True
    if numero > 0 and numero <= 10:
      ok = True
    else:
      print ('O número deve ser entre 1 a 10. Tente novamente.')
      ok = False
  except ValueError:
    print ('O número deve ser inteiro. Tente novamente.')

print (''*2)
print (f'Tabuada de {numero}:')
print (''*2)
print (f'{numero} X 1  = {numero*1}') 
print (f'{numero} X 2  = {numero*2}')
print (f'{numero} X 3  = {numero*3}')
print (f'{numero} X 4  = {numero*4}')
print (f'{numero} X 5  = {numero*5}')
print (f'{numero} X 6  = {numero*6}')
print (f'{numero} X 7  = {numero*7}')
print (f'{numero} X 8  = {numero*8}')
print (f'{numero} X 9  = {numero*9}')
print (f'{numero} X 10 = {numero*10}')

Digite um número inteiro de 1 a 10: 48
O número deve ser entre 1 a 10. Tente novamente.
Digite um número inteiro de 1 a 10: 67.4
O número deve ser inteiro. Tente novamente.
Digite um número inteiro de 1 a 10: 9

Tabuada de 9:

9 X 1  = 9
9 X 2  = 18
9 X 3  = 27
9 X 4  = 36
9 X 5  = 45
9 X 6  = 54
9 X 7  = 63
9 X 8  = 72
9 X 9  = 81
9 X 10 = 90


**Exercício 13.**
* Faça um programa que peça dois números, base e expoente, calcule e mostre o primeiro número elevado ao segundo número. 
 * Não utilize a função de potência da linguagem.

In [ ]:
print ('Calculando as potências.')
print (' '*2)

#BASE
ok = False
while not ok:
  try:
    base = int(input('Digite o número base: '))
    ok = True
  except ValueError:
    print ('O número deve ser inteiro. Tente novamente.')

#EXPOENTE
ok = False
while not ok:
  try:
    expoente = int(input('Digite o número expoente: '))
    ok = True
  except ValueError:
    print ('O número deve ser inteiro. Tente novamente.')

#RESULTADO
print (' '*2)
print (f'O resultado de {base}**{expoente} é: {base ** expoente}')

Calculando as potências.
  
Digite o número base: dasdeas
O número deve ser inteiro. Tente novamente.
Digite o número base: 3.4
O número deve ser inteiro. Tente novamente.
Digite o número base: 3
Digite o número expoente: dades
O número deve ser inteiro. Tente novamente.
Digite o número expoente: 4.6
O número deve ser inteiro. Tente novamente.
Digite o número expoente: 6
  
O resultado de 3**6 é: 729


**Exercício 14.**
* Faça um programa que peça 10 números inteiros, calcule e mostre a quantidade de números pares e a quantidade de números impares.

In [ ]:
posicao = 1
par = 0
impar = 0
while posicao <= 10:
  entrada = int(input('Digite um número: '))
  posicao = posicao + 1
  if entrada % 2 == 0:
    par = par + 1
  else:
    impar = impar + 1

print (''*2)
print(f'A quantidade de números pares é: {par}')
print(f'A quantidade de números ímpares é: {impar}')

Digite um número: 2
Digite um número: 4
Digite um número: 8
Digite um número: 10
Digite um número: 1
Digite um número: 3
Digite um número: 5
Digite um número: 99
Digite um número: 57
Digite um número: 3

A quantidade de números pares é: 4
A quantidade de números ímpares é: 6


**Exercício 15.**

* A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um programa capaz de gerar a série até o n−ésimo termo.


#FUNÇÕES PYTHON

In [ ]:
def titulo(txt):
  print ('-'*30)
  print (txt)
  print ('-'*30)

titulo('Olá Mundo')

############################
def soma (a,b):
  s = a * b
  print (f'A área do terreno é de {s}m².')
  print ('-'*30)

largura = float (input('Digite quantos metros de largura tem o terreno: '))
comprimento = float (input('Digite quantos metros de comprimento tem o terreno: '))

soma (largura,comprimento)
############################



------------------------------
Olá Mundo
------------------------------
Digite quantos metros de largura tem o terreno: 15
Digite quantos metros de comprimento tem o terreno: 2.5
A área do terreno é de 37.5m².
------------------------------


In [ ]:
# SEGUNDO MAIOR - MODELO BASICO #

def contador (lst):
  tam = len(lst)
  print (f'Têm {tam} números na lista.')
  maior = max(lst)
  print (f'O maior número da lista é o {maior}')
  penultimo = lst.remove(maior)
  penultimo = max(lst)
  print (f'O penultimo maior número da lista é o {penultimo}')

lista = [9, 7, 8, 1012, 151, 52, 190, 95, 109]
contador (lista)

Têm 9 números na lista.
O maior número da lista é o 1012
O penultimo maior número da lista é o 190


In [ ]:
# MAIOR NÚMERO EM UMA LISTA #

lista = [9, 7, 8, 102, 1510, 5245, 190, 95, 109]

def maior(lista):
  maior_numero = float('-inf')
  for valor in lista:
    if valor > maior_numero:
      maior_numero = valor  
  return maior_numero

print(maior(lista))

5245


#[Exercícios Listas](https://wiki.python.org.br/ExerciciosListas)
* Em andamento...



**Exercício 1.**

**Faça um Programa que leia um vetor de 5 números inteiros e mostre-os.**


In [ ]:
#first step: cria-se uma lista vazia com valor vazio.
#função de acrescentar um elemento ao final da lista.
# lista.append()
#cria-se um laço de repetição "for" com adesão do "_"
#pelo fato de não precisar saber o número que será adicionado ao vetor.

lista = []
for _ in range(5): # repetirá o laço 5x.
  numero = float (input('Digite um número: '))
  lista.append(numero)
print (lista)

Digite um número: 3
Digite um número: 4
Digite um número: 6
Digite um número: 99
Digite um número: 95
[3.0, 4.0, 6.0, 99.0, 95.0]


**Exercício 2.**

**Faça um Programa que leia um vetor de 10 números reais e mostre-os na ordem inversa.**

In [ ]:
lista = []
for _ in range(10):
  numero = float( input(' Digite um número: '))
  lista.append(numero)

print(lista[::-1])

 Digite um número: 1
 Digite um número: 2
 Digite um número: 44
 Digite um número: 54
 Digite um número: 34
 Digite um número: 53
 Digite um número: 3
 Digite um número: 4
 Digite um número: 5
 Digite um número: 10
[10.0, 5.0, 4.0, 3.0, 53.0, 34.0, 54.0, 44.0, 2.0, 1.0]


**Exercício 3.**
**Faça um Programa que leia 4 notas, mostre as notas e a média na tela.**2

In [ ]:
lista = []
for _ in range(4):
  numero = float(input('Digite o valor da nota: '))
  lista.append(numero)
  soma_lista = 0
  for i in lista:
    soma_lista += i
    media_lista = soma_lista/4
print(' '*2)
print('As notas foram:')
for i in lista:
  print(i)
print(f"Logo, a média alcançada foi {media_lista}")

Digite o valor da nota: 1
Digite o valor da nota: 2
Digite o valor da nota: 3
Digite o valor da nota: 4
  
As notas foram:
1.0
2.0
3.0
4.0
Logo, a média alcançada foi 2.5


**Exercício 4.**

**Faça um Programa que leia um vetor de 10 caracteres, e diga quantas consoantes foram lidas. Imprima as consoantes.**

In [ ]:
#VALIDADOR DE 10 CARACTERES.
ok = False
while not ok:
  try:
    palavra = str(input('Digite uma palavra com 10 caracteres: '))
    ok = True
    if len(palavra) >= 10:
      ok = True
    else:
      print ('A palavra deve conter 10 caracteres. Tente novamente.')
      ok = False
  except ValueError:
    print ('Você deve digitar somente letras. Tente novamente.')
else:
  resposta = (f'A palavra digitada foi {palavra}. ')
  print ('-'*len(resposta))
  print (resposta)
  print (f'Tendo no total {len(palavra)} caracteres.')
  print ('-'*len(resposta))

#CONTADOR DE VOGAIS E CONSOANTES.
vogais = 0
consoantes = 0
for caracter in palavra:
  if caracter in 'aeiou':
    vogais = vogais + 1
  else:
    consoantes = consoantes + 1

print(f'Essa palavra tem {vogais} vogais.')
print(f'Essa palavra tem {consoantes} consoantes.')

Digite uma palavra com 10 caracteres: luizcarlos
-----------------------------------
A palavra digitada foi luizcarlos. 
Tendo no total 10 caracteres.
-----------------------------------
Essa palavra tem 4 vogais.
Essa palavra tem 6 consoantes.


**Exercício 5.**

**Faça um Programa que leia 20 números inteiros e armazene-os num vetor. Armazene os números pares no vetor PAR e os números IMPARES no vetor impar. Imprima os três vetores.**


In [ ]:
lista = []
pares = []
impares = []

### VALIDADOR NUMERO INTEIRO ###
def leiaInt (mensagem):
  ok = False
  valor = 0
  while True:
    numero = str(input(mensagem))
    if numero.isnumeric():
      valor = int(numero)
      ok = True
    else:
      print ('O número não é inteiro. Tente novamente.')
    if ok:
      break
  return valor  

for _ in range(20):
  numero = leiaInt('Digite um número inteiro: ')
  lista.append(numero)
  if numero %2 == 0:
    pares.append(numero)
  else:
    impares.append(numero)

print (''*2)
print ('Números acrescentados à lista:')
print (lista)
print (''*2)
print (f'Foram incluídos {len(pares)} números pares:')
print (pares)
print (''*2)
print (f'E {len(impares)} números impares:')
print (impares)

#Script:
#criar 3 listas vazias: lista, pares, impares
#input dos numeros (+ validador *opcional)
#append lista 
#if append pares
#else append impares
#impressao de lista
#impressao de pares
#impressao de impares


Digite um número inteiro: luiz
O número não é inteiro. Tente novamente.
Digite um número inteiro: 1.4
O número não é inteiro. Tente novamente.
Digite um número inteiro: 1
Digite um número inteiro: 2
Digite um número inteiro: 3
Digite um número inteiro: 4
Digite um número inteiro: 5
Digite um número inteiro: 6
Digite um número inteiro: 7
Digite um número inteiro: 8
Digite um número inteiro: 9
Digite um número inteiro: 10
Digite um número inteiro: 11
Digite um número inteiro: 12
Digite um número inteiro: 13
Digite um número inteiro: luiz
O número não é inteiro. Tente novamente.
Digite um número inteiro: 14
Digite um número inteiro: 15
Digite um número inteiro: 16
Digite um número inteiro: 17
Digite um número inteiro: 18
Digite um número inteiro: 19
Digite um número inteiro: 20

Números acrescentados à lista:
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]

Foram incluídos 10 números pares:
[2, 4, 6, 8, 10, 12, 14, 16, 18, 20]

E 10 números impares:
[1, 3, 5, 7, 9,

**Exercício 6.**

**Faça um Programa que peça as quatro notas de 10 alunos, calcule e armazene num vetor a média de cada aluno, imprima o número de alunos com média maior ou igual a 7.0.**


In [ ]:
lista = []
acima_de_7 = []

for i in range(4): #reduzi o numero de alunos para teste.
  print (' '* 2)
  nome_aluno = input('Qual o nome do aluno? ')
  nota1 = float(input('Qual foi sua primeira nota? '))
  nota2 = float(input('Qual foi sua segunda nota? '))
  nota3 = float(input('Qual foi sua terceira nota? '))
  nota4 = float(input('Qual foi sua quarta nota? '))
  soma = nota1 + nota2 + nota3 + nota4
  media = soma/4
  if media >= 7:
    acima_de_7.append(media)
    
print (' '* 2)
print (f'O total de alunos que alcaçaram média maior ou igual a 7 foi: {len(acima_de_7)}')

  
Qual o nome do aluno? Luiz
Qual foi sua primeira nota? 10
Qual foi sua segunda nota? 8
Qual foi sua terceira nota? 7.5
Qual foi sua quarta nota? 9
  
Qual o nome do aluno? Carlos
Qual foi sua primeira nota? 9
Qual foi sua segunda nota? 6.5
Qual foi sua terceira nota? 4
Qual foi sua quarta nota? 2.5
  
Qual o nome do aluno? Pontes
Qual foi sua primeira nota? 10
Qual foi sua segunda nota? 5.75
Qual foi sua terceira nota? 6
Qual foi sua quarta nota? 4
  
Qual o nome do aluno? Gouveia
Qual foi sua primeira nota? 10
Qual foi sua segunda nota? 10
Qual foi sua terceira nota? 9
Qual foi sua quarta nota? 8.5
O total de alunos que alcaçaram média maior ou igual a 7 foi: 2


**Exercício 7.**

**Faça um Programa que leia um vetor de 5 números inteiros, mostre a soma, a multiplicação e os números.**


In [ ]:
lista = []
soma_lista = []
multiplica_lista = []

for _ in range(5):
  numero = int(input('Digite um número inteiro: '))
  lista.append(numero)
print (' '*2)
print(f'Os valores acrescentados foram: {lista}')

soma = 0
for i in lista:
  soma+= i
print (' '*2)
print (f'A soma desses numeros é = {soma}')
#soma_lista.append(soma)

multiplica = 1
for i in lista:
  multiplica*= i
print (' '*2)
print (f'Se multiplicarmos todos eles o resultado é = {multiplica}')
#multiplica_lista.append(multiplica)


Digite um número inteiro: 4
Digite um número inteiro: 7
Digite um número inteiro: 10
Digite um número inteiro: 5
Digite um número inteiro: 8
  


SyntaxError: ignored

**Exercício 8**

**Faça um Programa que peça a idade e a altura de 5 pessoas, armazene cada informação no seu respectivo vetor. Imprima a idade e a altura na ordem inversa a ordem lida.**


In [ ]:
idades = []
alturas = []

for _ in range(5):
  print (''*2)
  idade= int(input('Qual a idade dessa pessoa: '))
  idades.append(idade)
  altura= float(input('Qual a altura dessa pessoa? '))
  alturas.append(altura)

print (idades[::-1])
print (alturas[::-1])


Qual a idade dessa pessoa: 32
Qual a altura dessa pessoa? 1.69

Qual a idade dessa pessoa: 33
Qual a altura dessa pessoa? 1.89

Qual a idade dessa pessoa: 56
Qual a altura dessa pessoa? 1.62

Qual a idade dessa pessoa: 18
Qual a altura dessa pessoa? 1.76

Qual a idade dessa pessoa: 22
Qual a altura dessa pessoa? 1.66
[22, 18, 56, 33, 32]
[1.66, 1.76, 1.62, 1.89, 1.69]


**Exercício 9.**

**Faça um Programa que leia um vetor A com 10 números inteiros, calcule e mostre a soma dos quadrados dos elementos do vetor.**


In [ ]:
vetor_a = []
vetor_a_quadrado = []
vetor_a_soma = []

for _ in range(10):
  numero = int(input('Digite um número inteiro: '))
  vetor_a.append(numero)

quadrado = 1
for n in vetor_a:
  quadrado = n**2
  vetor_a_quadrado.append(quadrado)

soma = 0
for s in vetor_a_quadrado:
  soma+= s

#Resposta
print(' '*2)
print(f'O quadrado de cada número é = {vetor_a_quadrado}')
print(f'A soma dos 10 números é = {soma}')


Digite um número inteiro: 3
Digite um número inteiro: 5
Digite um número inteiro: 6
Digite um número inteiro: 7
Digite um número inteiro: 9
Digite um número inteiro: 10
Digite um número inteiro: 8
Digite um número inteiro: 24
Digite um número inteiro: 52
Digite um número inteiro: 23
  
O quadrado de cada número é = [9, 25, 36, 49, 81, 100, 64, 576, 2704, 529]
A soma dos 10 números é = 4173


**Exercício 10.**

**Faça um Programa que leia dois vetores com 10 elementos cada. Gere um terceiro vetor de 20 elementos, cujos valores deverão ser compostos pelos elementos intercalados dos dois outros vetores.**


In [ ]:
vetor_a = []
vetor_b = []
intercalada = []

for _ in range(3):
  elementos_a = input('Adicione elementos ao VETOR A: ')
  vetor_a.append(elementos_a)

for n in range(3):
  elementos_b = input('Adicione elementos ao VETOR B: ')
  vetor_b.append(elementos_b)

print ("Os valores salvos no VETOR A foram: " + str(vetor_a)) 
print ("Os valores salvos no VETOR B foram: " + str(vetor_b)) 
juncao = vetor_a + vetor_b 
juncao[::2] = vetor_a 
juncao[1::2] = vetor_b 
print ("Os valores intercalados dos VETORES A + B são: " + str(juncao)) 

Adicione elementos ao VETOR A: 1
Adicione elementos ao VETOR A: 2
Adicione elementos ao VETOR A: 3
Adicione elementos ao VETOR B: 4
Adicione elementos ao VETOR B: 5
Adicione elementos ao VETOR B: 6
Os valores salvos no VETOR A foram: ['1', '2', '3']
Os valores salvos no VETOR B foram: ['4', '5', '6']
Os valores intercalados dos VETORES A + B são: ['1', '4', '2', '5', '3', '6']


**Exercício 11.**

**Altere o programa anterior, intercalando 3 vetores de 10 elementos cada.**


In [ ]:
vetor_a = []
vetor_b = []
vetor_c = []

for _ in range(3):
  elementos_a = input('Adicione elementos ao VETOR A: ')
  vetor_a.append(elementos_a)
print (' '*2)

for n in range(3):
  elementos_b = input('Adicione elementos ao VETOR B: ')
  vetor_b.append(elementos_b)
print (' '*2)

for z in range(3):
  elementos_c = input('Adicione elementos ao VETOR C: ')
  vetor_c.append(elementos_c)

print (' '*2)
print ("Os valores salvos no VETOR A foram: " + str(vetor_a)) 
print ("Os valores salvos no VETOR B foram: " + str(vetor_b))
print ("Os valores salvos no VETOR C foram: " + str(vetor_c)) 
juncao = vetor_a + vetor_b + vetor_c
juncao[::3] = vetor_a
juncao[1::3] = vetor_b 
juncao[2::3] = vetor_c
print ("Os valores intercalados dos VETORES A + B + C são: " + str(juncao)) 

Adicione elementos ao VETOR A: 1
Adicione elementos ao VETOR A: 2
Adicione elementos ao VETOR A: 3
  
Adicione elementos ao VETOR B: 4
Adicione elementos ao VETOR B: 5
Adicione elementos ao VETOR B: 6
  
Adicione elementos ao VETOR c: 7
Adicione elementos ao VETOR c: 8
Adicione elementos ao VETOR c: 9
  
Os valores salvos no VETOR A foram: ['1', '2', '3']
Os valores salvos no VETOR B foram: ['4', '5', '6']
Os valores salvos no VETOR C foram: ['7', '8', '9']
Os valores intercalados dos VETORES A + B + C são: ['1', '4', '7', '2', '5', '8', '3', '6', '9']


**Exercício 12.**

**Foram anotadas as idades e alturas de 30 alunos. Faça um Programa que determine quantos alunos com mais de 13 anos possuem altura inferior à média de altura desses alunos.**
